spec/dry/types/compiler_spec.rb

```diff
@@ -217,18 +170,18 @@
   end
 
   it 'builds an schema-less form.hash' do
-    ast = [:type, ['form.hash']]
+    ast = Dry::Types['form.hash'].schema([]).to_ast
 
     type = compiler.(ast)
 
-    expect(type[nil]).to be(nil)
+    expect(type[nil]).to eql({})
```

Comment:
This was the only spec I could not fulfill as it was, the coercion was returning always a `hash`, to code that makes this condition to happen is here https://github.com/dry-rb/dry-types/blob/type-ast/lib/dry/types/hash/schema.rb#L101-L102 since is an empty schema it did not have any `members` I don't have that concrete knowledge of the desired behavior